pkg/operator/endpoint_status_builder.go

```diff
@@ -68,24 +68,155 @@ func (b *scrapeEndpointBuilder) add(target *prometheusv1.TargetsResult) error {
 	return nil
 }
 
+type ScrapeJobKind uint8
+
+const (
+	ScrapeJobInvalid ScrapeJobKind = iota
+	ScrapeJobKubelet
+	ScrapeJobPodMonitoring
+	ScrapeJobClusterPodMonitoring
+	ScrapeJobNodeMonitoring
+)
+
+func probeKindToString(kind ScrapeJobKind) string {
+	switch kind {
+	case ScrapeJobKubelet:
+		return "kubelet"
+	case ScrapeJobPodMonitoring:
+		return "PodMonitoring"
+	case ScrapeJobClusterPodMonitoring:
+		return "ClusterPodMonitoring"
+	case ScrapeJobNodeMonitoring:
+		return "NodeMonitoring"
+	default:
+		return ""
+	}
+}
+
+func parseScrapeKind(kind string) (ScrapeJobKind, error) {
+	switch kind {
+	case probeKindToString(ScrapeJobKubelet):
+		return ScrapeJobKubelet, nil
+	case probeKindToString(ScrapeJobPodMonitoring):
+		return ScrapeJobPodMonitoring, nil
+	case probeKindToString(ScrapeJobClusterPodMonitoring):
+		return ScrapeJobClusterPodMonitoring, nil
+	case probeKindToString(ScrapeJobNodeMonitoring):
+		return ScrapeJobNodeMonitoring, nil
+	default:
+		return ScrapeJobInvalid, fmt.Errorf("invalid scrape kind %q", kind)
+	}
+}
+
+func parseScrapeJob(pool string) (monitoringv1.PodMonitoringCRD, error) {
+	split := strings.Split(pool, "/")
+	kind, err := parseScrapeKind(split[0])
+	if err != nil {
+		return nil, err
+	}
+	switch kind {
+	case ScrapeJobKubelet:
+		if len(split) != 1 {
+			return nil, fmt.Errorf("invalid Kubelet scrape job format %q", pool)
+		}
+		return nil, nil
+	case ScrapeJobPodMonitoring:
+		if len(split) != 3 {
+			return nil, fmt.Errorf("invalid PodMonitoring scrape job format %q", pool)
+		}
+		return &monitoringv1.PodMonitoring{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: split[1],
+				Name:      split[2],
+			},
+		}, nil
+	case ScrapeJobClusterPodMonitoring:
+		if len(split) != 2 {
+			return nil, fmt.Errorf("invalid ClusterPodMonitoring scrape job format %q", pool)
+		}
+		return &monitoringv1.ClusterPodMonitoring{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: split[1],
+			},
+		}, nil
+	case ScrapeJobNodeMonitoring:
+		if len(split) != 2 {
+			return nil, fmt.Errorf("invalid NodeMonitoring scrape job format %q", pool)
+		}
+		return nil, nil
+	default:
+		return nil, fmt.Errorf("unknown scrape kind %q", split[0])
+	}
+}
+
+type scrapePool struct {
+	job   string
+	group string
+}
+
+func parseScrapePool(pool string) (scrapePool, error) {
+	split := strings.Split(pool, "/")
+	kind, err := parseScrapeKind(split[0])
+	if err != nil {
+		return scrapePool{}, err
+	}
+	switch kind {
+	case ScrapeJobKubelet:
+		if len(split) != 2 {
+			return scrapePool{}, fmt.Errorf("invalid PodMonitoring scrape pool format %q", pool)
+		}
+		return scrapePool{
+			job:   split[0],
+			group: split[1],
+		}, nil
+	case ScrapeJobPodMonitoring:
+		if len(split) != 4 {
+			return scrapePool{}, fmt.Errorf("invalid PodMonitoring scrape pool format %q", pool)
+		}
+		index := len(split[0]) + len(split[1]) + len(split[2]) + 2
```

Comment:
why is it +2?

Reply:
Added comments. Let me know if it's unclear.

Reply:
Thanks, it's clearer now!